tuxemon/menu/menu.py

```diff
@@ -67,6 +68,7 @@ class Menu(Generic[T], state.State):
     default_character_delay = 0.05
     shrink_to_items = False  # fit the border to contents
     escape_key_exits = True  # escape key closes menu
+    back_state = None  # if escape_key_exits is false, then the escape key can change the state
```

Comment:
i don't see why the esc key and also back_state logic needs to be linked.  why should it not be possible to press escape and also use back_state?

Reply:
They should be linked since I programmed the menu to go to the back_state using the escape key, which escape_key_exits also uses as well.
Should I make it that when the `back_state is not None`, it'll go to the back_state no matter what value escape_key is?

Reply:
have you considered the discord comment i left?  we can actually do this without even adding the back_state
```
diff --git a/tuxemon/states/start/__init__.py b/tuxemon/states/start/__init__.py
index ebbb2b66..a9d574fc 100644
--- a/tuxemon/states/start/__init__.py
+++ b/tuxemon/states/start/__init__.py
@@ -90,17 +90,18 @@ class StartState(PopUpMenu[StartGameObj]):
             )
 
         def set_player_name(text: str) -> None:
+            map_name = prepare.fetch("maps", prepare.CONFIG.starting_map)
+            self.client.push_state("WorldState", map_name=map_name)
+            self.client.push_state(FadeInTransition)
             local_session.player.name = text
 
         def new_game() -> None:
             # load the starting map
-            map_name = prepare.fetch("maps", prepare.CONFIG.starting_map)
-            self.client.replace_state("WorldState", map_name=map_name)
             self.client.push_state(
                 state_name=InputMenu,
                 prompt=T.translate("input_name"),
                 callback=set_player_name,
-                escape_key_exits=False,
+                escape_key_exits=True,
             )
             self.client.push_state(FadeInTransition)
```
the benefit here is no new added features, which, honestly we kinda try to avoid since it just adds a maintenance burden.
to answer your question though, i don't see why they should be linked.  the back key should only close the state.  adding a conditional to it (close the state?  close and use the back state?) is confusing because in both instances, the menu state is closed anyway.  and the existence of a back state is will also enable the back state.  to put it another way, it wouldn't make sense to have a situation where there is a back_state and also disabling the escape key.

Reply:
Sorry, I didn't see the negatives of adding a new feature. I added your pieces of code to the pull request and resolved some of your other comments.